README.md
# skytest-official-release
skytest version 1.0 official release


changelog.txt
--------------------1.4--------------------

- added Growth essence to instant grow any of the ore crops
-tweaked growth rates of ore papy
-added fountain looking infinite water source. right click with any empty bucket and get infinite water
-added blank seeds and ore plants just like magical crops in minecraft
-ore papyrus now drops itself wich then needs to be crafted in a 3x3 to get respective ore, super sprouts still drop ore
-fixed transparency on hammer textures 
-diamond hammer mines sand and gravel the same speed as a steel shovel
-added copper and tin ore chunks
-fixed barrels barrels and crucibles top texture always showing oak wood
-added world ancor(will keep a portion of the world always loaded)
-clay bucket is now compatable with barrels and crucibles
-added lapis,redstone, glowstone, and tin, infused dirt and super infused dirt
-various tweaks to sieves and machines now run smoother 
-added clay bucket for getting lava from crucible
-added climing pick that replicates the sneak glitch 
-added engraver(only works for 0.4.16 and newer) right click with any item to change the discription of it.(like Minecraft anvil)
-added achievements for most of the ones in the achievement book to detect and update automaticly
-added freezer that can turn water into ice...nice. will also cool lava 
-added working formspec bassed tetris game to play. Perfect for me cus we all know I have no friends :(
-added working formspec bassed chess game to play with friends. Usless if your alone like me :(
-just for fun, nyan cats will now move leaving a trail of rainbows, if on ehits an unloaded chunk or a block besids air it will change direction
-added chisel to chisel blocks into differant shapes
-chisel has 7 modes{
#1:Chisel 4 Edges
#2:Horizontal Groove
#3:Vertical Groove
#4:Cross Grooves
#5:Square
#6:Chisel ladder
#7:Chisel Foot Hold
}
-added super bright star light that produces far more than 14 light level
-various bug fixes
-various new crafting items and recipies for making stuff
-removed enirboreh

--------------------1.3--------------------

-added angel block with wich one can fly and use to place a block under your feet
-added oak, aspen, pine, jungle, and acacia wooden barrels(all work exactly the same)
-if extra_util mod is enabled(another of my mods, currently unreleased){ 
-dsu cannot hold tools
-added DSU(Deep Storage Unit)holds 2,000,000 of ONE item. can be transporteed by using the formspec(dont break of you will lose all items in it.)crafted with bedrockium ingots and 1 enchanted block
}
-if 3d_armor mod is enabled{ 
-added a pair o' wings(grants the fly priv. while wearing them)
}
-if pipeworks mod is enabled{ 
-added pipe support for DSU, all drawers(all woods + 1x1 1x2 2x2 drawers)
-if using pipeworks + mesecons you can automate pretty much every thing by using deployers and node breakers. deployers will prety much rightclick on any block with what every is in theier inventory. nodebreakers will break nodes...BUT you can insert any tool and it will yeild the same results as if you used the same tool on the node, meaning you can use hammers, crooks, silkworms, ect.
}
-if MCL_core mod is enabled{ 
-added support for all types of wood for drawers
}
-added achievement book with a few goals in it already(more to come)
-added x1 x2 and x4 storage drawers for all wood types
-all leaf colectors can be used on infested leaves to collect them
-added enchanted crystal and enchanted block(used to craft stuff)
-added entity jetpack( use arow keys, spacebar, and shift key to control)warnign is effected by lag, fly at your own risk
-silkworms can be cooked and eaten(If you have a stomoch of iron)
-rightclick on a wooden barrel that is full of water with dust and the barrel will fill with clay. Rightclick to retrieve the clay block
-added Mk2 cobble gen
-added doc support for (almost) every item / block/ tool that actuly does something
-removed dense leaves
-removed leaf press(the one with the gui/inventory/formspec and ABM)
-use a leaf weight on a wooden barrel then fill with either cactus of "group:leaves" and wait (after 32 seconds mit will extract the water frop the items, at wich point using a wooden or iron bucket will be fill the with water)  
-removed compost bin gui/inventory/formspec
-replaced compostbin with wooden barrel(the same items can be composted)barrel will work just like the one in minecraft(used to get dirt, water, clay, ect.)
-wooden barrel has a decomposing animation
-added all the colors of mulch(blue, green, ect.)mulch is affected by gravity
-added advanced compost bin(like regular but can get you clay and mulch as well, also works instantly and can color mulch)
-added porcelain lumps(use to craft raw crucible)
-added Raw crucible(smelt into fired crucible)
-added Fired crucible(input 8 cobble and place a torch or anything from group:igniter directly under to slowly melt the cobble into lava) igniter will work faster than torch. when the lava is done use a iron bucket
-added an uncraftable item for testing that tells you how many registerd items nodes and tools there are in that world(lists all strings in items file in world folder)
-removed auto sieves' gui/inventory/formspec
-auto sieve now only sieves faster than normal sieve
-auto sieve drops 2X that of normal sieve
-normal sieve is now a multi block structure(just like minecraft)
-sieves can now drop these items{
listed rarest to most common
dirt will drop 1 item and 1 pebble
dirt:
"skytest:strawberrybush_seeds"
"skytest:raspberrybush_seeds"
"skytest:pine_tree_seeds"
"skytest:pebble 7"
"skytest:acacia_tree_seeds"
"skytest:aspen_tree_seeds"
"skytest:oak_tree_seeds"
"farming_plus:banana_sapling"
"farming_plus:cocoa_sapling"
"skytest:jungle_tree_seeds"
"skytest:cactus_seeds"
"skytest:papyrus_seeds"
"farming_plus:tomato_seed"
"farming_plus:rhubarb_seed"
"skytest:pebble 3"
"farming_plus:carrot_seed"

sieve.lua
local percentdirt = {
    {"skytest:strawberrybush_seeds", 1},
    {"skytest:raspberrybush_seeds", 2},
    {"skytest:pine_tree_seeds", 3},
    {"skytest:pebble 7", 4},
    {"skytest:acacia_tree_seeds", 5},
    {"skytest:aspen_tree_seeds", 6},
    {"skytest:oak_tree_seeds", 7},
    {"farming_plus:banana_sapling", 8},
    {"farming_plus:cocoa_sapling", 9},
    {"skytest:jungle_tree_seeds", 10},
    {"skytest:cactus_seeds", 11},
    {"skytest:papyrus_seeds", 12},
    {"farming_plus:tomato_seed", 13},
    {"farming_plus:rhubarb_seed", 14},
    {"skytest:pebble 3", 16},
    {"farming_plus:carrot_seed", 17},
    {"crops:tomato_seed", 18},
    {"crops:green_bean_seed", 19},
    {"crops:potato_eyes", 20},
    {"crops:melon_seed", 20},
    {"crops:pumpkin_seed", 22},
    {"crops:corn", 23},
    {"skytest:pebble 2", 24},
    {"morefarming:seed_corn", 25},
    {"farming_plus:strawberry_seed", 26},
    {"farming_plus:orange_seed", 27},
    {"farming_plus:potato_seed", 28},
    {"skytest:grass_seeds", 29},
    {"farming:seed_wheat", 30},
    {"farming:seed_cotton", 31},
    {"morefarming:seed_wildcarrot", 32},
    {"morefarming:seed_teosinte", 33},
}
local percentgravel = {
    {"skytest:diamond_chunks", 5},
    {"default:mese_crystal_fragment 3", 7},
    {"skytest:gold_ore_chunks", 9},
    {"skytest:lapis", 11},
    {"skytest:redstone", 13},
    {"skytest:glowstone", 15},
    {"skytest:tin_ore_chunks", 17},
    {"skytest:copper_ore_chunks", 19},
    {"skytest:iron_ore_chunks", 24},
    {"default:coal_lump", 28},
    {"default:flint", 34},
}
function gravel()
    local chance = math.random(40)
    for _,v in pairs(percentgravel) do
        if chance<v[2] then
            return ItemStack(v[1])
        end
    end
    return nil
end
function dirt()
    local chance = math.random(35)
    for _,v in pairs(percentdirt) do
        if chance<v[2] then
            return ItemStack(v[1])
        end
    end
    return nil
end
minetest.register_node("skytest:hand_sieve", {
    description = "Hand Sieve",
    _doc_items_usagehelp = "Use to manuly sift dirt, sand, gravel, or dust. Dirt will always drop 1 pebble and may drop one item from this list(rarest to most common):(skytest:strawberrybush_seeds,skytest:raspberrybush_seeds, skytest:pine_tree_seeds, skytest:pebble 7, skytest:acacia_tree_seeds, skytest:aspen_tree_seeds,skytest:oak_tree_seeds,farming_plus:banana_sapling, farming_plus:cocoa_sapling,skytest:jungle_tree_seeds,skytest:cactus_seeds,skytest:papyrus_seeds,farming_plus:tomato_seed, farming_plus:rhubarb_seed, skytest:pebble 3,farming_plus:carrot_seed, crops:tomato_seed, crops:green_bean_seed,crops:potato_eyes, crops:melon_seed, crops:pumpkin_seed, crops:corn,skytest:pebble 2, morefarming:seed_corn, farming_plus:strawberry_seed, farming_plus:orange_seed,  farming_plus:potato_seed, skytest:grass_seeds, farming:seed_wheat, farming:seed_cotton, morefarming:seed_wildcarrot, morefarming:seed_teosinte) gravel will drop 1 item from this list(listed rarest to most common):(skytest:diamond_chunks,default:mese_crystal_fragment 3,skytest:gold_chunks,skytest:lapis,skytest:redstone,skytest:glowstone,skytest:iron_chunks,default:copper_lump,default:coal_lump,default:flint) sand will drop 2 items from the previous list, dust will drop 3. Growth crystals can be sieved and are equevelent to 2 gravel and 2 dirt.",
    paramtype = "light",
    tiles = {
    	"sieve_hand_sieve_top.png",
    	"sieve_hand_sieve_bottom.png",
    	"default_wood.png",
    	"default_wood.png",
    	"default_wood.png",
    	"default_wood.png"
    },
    groups = {oddly_breakable_by_hand=1},
    drawtype = "nodebox",
    node_box = {
        type = "fixed",
        fixed = {
            {-0.5, 0.1875, -0.5, 0.5, 0.19, 0.5}, --top
            {0.5,0.1875,0.5,-0.5,0.5,0.4375},--top
            {-0.5,0.1875,0.5,-0.4375,0.5,-0.5}, --top
            {0.5,0.1875,-0.5,-0.5,0.5,-0.4375}, --top
            {0.5,0.1875,0.5,0.4375,0.5,-0.5}, --top
            {-0.5, -0.5, -0.5, -0.4375, 0.5, -0.4375}, --leg
            {-0.5, -0.5, 0.5, -0.4375, 0.5, 0.4375}, --leg
            {0.5, -0.5, 0.5, 0.4375, 0.5, 0.4375}, --leg
            {0.5, -0.5, -0.5, 0.4375, 0.5, -0.4375}, --leg
        },
    },
    on_rightclick = function(pos, node, player, itemstack, pointed_thing)
        local timer = minetest.get_node_timer(pos)
        if itemstack:get_name() == "skytest:dust" and not timer:is_started() then
            pos.y = pos.y + 0.5
	    itemstack:take_item(1)
            timer:start(2)
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            return itemstack
         end
      	if itemstack:get_name() == "default:dirt" and not timer:is_started() then
            pos.y = pos.y + 0.5
	    itemstack:take_item(1)
            timer:start(2)
            minetest.add_item(pos, dirt())
	    minetest.add_item(pos, "skytest:pebble")
            return itemstack
  	end
	if itemstack:get_name() == "default:gravel" and not timer:is_started() then
            pos.y = pos.y + 0.5
	    itemstack:take_item(1)
            timer:start(2)
            minetest.add_item(pos, gravel())
            return itemstack
  	end
	if itemstack:get_name() == "default:sand" and not timer:is_started() then
            pos.y = pos.y + 0.5
	    itemstack:take_item(1)
            timer:start(2)
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            return itemstack
  	end
        if itemstack:get_name() == "skytest:growth_crystal" and not timer:is_started() then
            pos.y = pos.y + 0.5
	    itemstack:take_item(1)
            timer:start(2)
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, dirt())
            minetest.add_item(pos, dirt())
            return itemstack
  	end
    end,
    drop = { 
        max_items = 1,
        items = {{
            items = {"skytest:empty_sieve","skytest:mesh"},
	},}
    }
})
minetest.register_node("skytest:empty_sieve", {
    description = "Empty Sieve",
    paramtype = "light",
    tiles = {
    	"default_wood.png",
    	"default_wood.png",
    	"default_wood.png",
    	"default_wood.png"
    },
    groups = {oddly_breakable_by_hand=1},
    drawtype = "nodebox",
    node_box = {
        type = "fixed",
        fixed = {
            {-0.5, 0.1875, -0.5, 0.5, 0.19, 0.5}, --top
            {0.5,0.1875,0.5,-0.5,0.5,0.4375},--top
            {-0.5,0.1875,0.5,-0.4375,0.5,-0.5}, --top
            {0.5,0.1875,-0.5,-0.5,0.5,-0.4375}, --top
            {0.5,0.1875,0.5,0.4375,0.5,-0.5}, --top
            {-0.5, -0.5, -0.5, -0.4375, 0.5, -0.4375}, --leg
            {-0.5, -0.5, 0.5, -0.4375, 0.5, 0.4375}, --leg
            {0.5, -0.5, 0.5, 0.4375, 0.5, 0.4375}, --leg
            {0.5, -0.5, -0.5, 0.4375, 0.5, -0.4375}, --leg
        },
    },
    on_rightclick = function(pos, node, player, itemstack, pointed_thing)
        if itemstack:get_name() == "skytest:mesh" then
            minetest.set_node(pos, {name="skytest:hand_sieve"})
	    itemstack:take_item(1)
            return itemstack
         end
    end,
})
minetest.register_node("skytest:auto_sieve", {
    description = "Auto Sieve",
    _doc_items_usagehelp = "Use to instantly sift dirt, sand, gravel, or dust. Dirt will always drop 1 pebble and may drop two item from this list(rarest to most common):(skytest:strawberrybush_seeds,skytest:raspberrybush_seeds, skytest:pine_tree_seeds, skytest:pebble 7, skytest:acacia_tree_seeds, skytest:aspen_tree_seeds,skytest:oak_tree_seeds,farming_plus:banana_sapling, farming_plus:cocoa_sapling,skytest:jungle_tree_seeds,skytest:cactus_seeds,skytest:papyrus_seeds,farming_plus:tomato_seed, farming_plus:rhubarb_seed, skytest:pebble 3,farming_plus:carrot_seed, crops:tomato_seed, crops:green_bean_seed,crops:potato_eyes, crops:melon_seed, crops:pumpkin_seed, crops:corn,skytest:pebble 2, morefarming:seed_corn, farming_plus:strawberry_seed, farming_plus:orange_seed,  farming_plus:potato_seed, skytest:grass_seeds, farming:seed_wheat, farming:seed_cotton, morefarming:seed_wildcarrot, morefarming:seed_teosinte) gravel will drop 2 item from this list(listed rarest to most common):(skytest:diamond_chunks,default:mese_crystal_fragment 3,skytest:gold_chunks,skytest:lapis,skytest:redstone,skytest:glowstone,skytest:iron_chunks,default:copper_lump,default:coal_lump,default:flint) sand will drop 4 items from the previous list, dust will drop 6. Growth crystals can be sieved and are equevelent to 4 gravel and 4 dirt.",
    paramtype = "light",
    tiles = {
        "sieve_auto_sieve_top.png",
        "sieve_auto_sieve_bottom.png",
        "sieve_auto_sieve_side.png",
        "sieve_auto_sieve_side.png",
        "sieve_auto_sieve_side.png",
        "sieve_auto_sieve_side.png"
    },
    is_ground_content = true,
    groups = {choppy = 2, oddly_breakable_by_hand = 2},
    drawtype = "nodebox",
    node_box = {
        type = "fixed",
        fixed = {
                {-0.5, 0.1875, -0.5, 0.5, 0.19, 0.5},
                {0.5,0.5,-0.5,-0.5,0,-0.4375},
                {-0.5,0.5,-0.5,-0.4375,0,0.5},
                {0.5,0.5,0.5,-0.5,0,0.4375},
                {0.5,0.5,-0.5,0.4375,0,0.5},
                {-0.5, -0.5, -0.5, -0.4375, 0.5, -0.4375},
                {-0.5, -0.5, 0.5, -0.4375, 0.5, 0.4375},
                {0.5, -0.5, 0.5, 0.4375, 0.5, 0.4375},
                {0.5, -0.5, -0.5, 0.4375, 0.5, -0.4375},
                },
        },
    on_rightclick = function(pos, node, player, itemstack, pointed_thing)
        if itemstack:get_name() == "skytest:dust" then
            pos.y = pos.y + 0.5
	    itemstack:take_item(1)
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            return itemstack
         end
      	if itemstack:get_name() == "default:dirt" then
            pos.y = pos.y + 0.5
	    itemstack:take_item(1)
            minetest.add_item(pos, dirt())
            minetest.add_item(pos, dirt())
	    minetest.add_item(pos, "skytest:pebble")
            return itemstack
  	end
	if itemstack:get_name() == "default:gravel" then
            pos.y = pos.y + 0.5
	    itemstack:take_item(1)
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            return itemstack
  	end
	if itemstack:get_name() == "default:sand" then
            pos.y = pos.y + 0.5
	    itemstack:take_item(1)
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            return itemstack
  	end
        if itemstack:get_name() == "skytest:growth_crystal" then
            pos.y = pos.y + 0.5
	    itemstack:take_item(1)
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, gravel())
            minetest.add_item(pos, dirt())
            minetest.add_item(pos, dirt())
            minetest.add_item(pos, dirt())
            minetest.add_item(pos, dirt())
            return itemstack
  	end
    end,
          
})
--Shaker Motor
minetest.register_craftitem("skytest:shaker_motor", {
    description = "Shaker Motor",
    inventory_image = "sieve_shaker_motor.png",
    })
--Shaker Frame
minetest.register_craftitem("skytest:shaker_frame", {
    description = "Shaker Frame",
    inventory_image = "sieve_shaker_frame.png",
    })
--Auto Top
minetest.register_craftitem("skytest:auto_sieve_top", {
    description = "Auto Sieve Top",
    inventory_image = "sieve_auto_top.png",
})
minetest.register_craft({
    output = "skytest:shaker_motor 2",
    recipe = {
        {"", "",""},
        {"default:cobble", "default:steel_ingot","default:cobble"},
        {"default:cobble", "default:copper_ingot","default:cobble"},
    }
})
minetest.register_craft({
    output = "skytest:shaker_frame 1",
    recipe = {
        {"default:steel_ingot", "skytest:shaker_motor","default:steel_ingot"},
        {"skytest:shaker_motor", "default:furnace","skytest:shaker_motor"},
        {"default:steel_ingot", "skytest:shaker_motor","default:steel_ingot"},
    }
})
--Auto Legs
minetest.register_craftitem("skytest:auto_sieve_legs", {
    description = "Auto Sieve Legs",
    inventory_image = "sieve_auto_legs.png",
})
minetest.register_craft({
    output = "skytest:auto_sieve_top",
    recipe = {
        {"default:steel_ingot", "default:steel_ingot", "default:steel_ingot"},
        {"default:steel_ingot", "skytest:mesh", "default:steel_ingot"},
        {"default:steel_ingot", "default:steel_ingot", "default:steel_ingot"}
    }
})
minetest.register_craft({
    output = "skytest:auto_sieve_legs",
    recipe = {
        {"default:steel_ingot", "skytest:shaker_frame", "default:steel_ingot"},
        {"default:steel_ingot", "", "default:steel_ingot"}
    }
})
minetest.register_craft({
    output = "skytest:auto_sieve",
    recipe = {
        {"skytest:auto_sieve_top"},
        {"skytest:auto_sieve_legs"}
    }
})